### DISCUSSION TITLE
How to generate document automation attach image link

### DISCUSSION THREAD

**Discussion by chuongmep** [dotnet/docfx#8469] [General]
**Describe the solution you'd like**
Hi team, at the moment, I need move this source to docfx, but I don't know how can do with same format also automation attacth image sample bellow function API, it is can do with docfx ? 
This is document I created and I want move to docfx: https://chuongmep.github.io/OpenMEP/element/element.html

Any comment appreciate !
![image](https://user-images.githubusercontent.com/31106432/222453274-629a25f8-9deb-4a60-bd09-cb84f6912f53.png)

**Comment by yufeih** [comment]
Does markdown image link in XML comment work?

**Comment by chuongmep** [comment]
@yufeih it is not working for me now, with one method, I need demo one image, so I can't create a lot **markdown** for that. Example : 
```
/// [!INCLUDE [Image](../../docs/OpenMEPPage/element/family/dyn/pic/Family.EditFamily.png)]
```
and then it is result : 
![image](https://user-images.githubusercontent.com/31106432/222718575-3b02d897-54a6-4592-a576-4ac2a76c0fa8.png)

I hope to have a guideline for single image like video in here : 
![image](https://user-images.githubusercontent.com/31106432/222721929-85a0df61-648f-43a1-a556-6d636ac7fbb6.png)

**Comment by yufeih** [comment]
Here is the markdown image syntax `![alt text](image.png "Title")`

**Reply by chuongmep** [reply]
Hi @yufeih, but problem with me is with 1000 method I need to attach 1000 images  sample, can we have option like embed video no need create 1000 markdown file ?

**Reply by yufeih** [reply]
I'm not sure I understand the question, you can put markdown directly in csharp xml comments instead of markdown files.

```csharp
/// <examples>
///  ![alt text](image.png "Title")
/// </examples>
```

**Reply by chuongmep** [reply]
Thanks for your @yufeih, I think documentation guideline missing guide for attach image. Thank you. 
https://dotnet.github.io/docfx/
https://dotnet.github.io/docfx/docs/markdown.html?tabs=linux%2Cdotnet

**Reply by chuongmep** [reply]
Hi @yufeih tried, but it is not working, do you have tried with relative path yet? 

```xml
 /// <summary>
    /// return current version of dynamo
    /// </summary>
    /// <returns name="version">version</returns>
    /// <example>
    /// ![alt text](../../docs/OpenMEPPage/application/dyn/pic/Dynamo.Version.png "Dynamo")
    /// </example>
```
![image](https://user-images.githubusercontent.com/31106432/223410956-8b5ed926-98fd-4619-b059-aaffbd65444f.png)

**Reply by yufeih** [reply]
The path is probably relative to the generated `.yml` file location, not the `.cs` file. Any warnings appear in the output?

**Reply by chuongmep** [reply]
Hi @yufeih , have a warning relate for that. This is project open source, can help me review for that ?  
- Project : https://github.com/chuongmep/OpenMEP
- Location of docfx.json : https://github.com/chuongmep/OpenMEP/blob/dev/docfx_project/docfx.json
- Location of method : https://github.com/chuongmep/OpenMEP/blob/db347bc0dca19b27148dfb3509fbd3bedd8a79ac/OpenMEP/Application/Dynamo.cs#L19
- Location of image : https://github.com/chuongmep/OpenMEP/tree/dev/docs/OpenMEPPage/application/dyn/pic
Warning : 
warning InvalidFileLink: Invalid file link:(~/../docs/OpenMEPPage/application/dyn/pic/Dyna
mo.Version.png).

**Reply by yufeih** [reply]
Aha, you hit the issue that docfx cannot reference stuff outside of the `docfx.json` folder, can the `docfx.json` be moved to `docs/docfx.json`?

**Reply by chuongmep** [reply]
@yufeih tried move but still same problem, this is a log : 

```
Using .NET Core SDK 6.0.402
Loading project C:/Users/chuon/RiderProjects/GlobalMEP/OpenMEP/OpenMEP.csproj
[Warning] Found project reference without a matching metadata reference: C:\Users\chuon\RiderProjects\GlobalMEP\OpenMEPViewExtension\OpenMEPViewExtension.csproj
[Warning] Found project reference without a matching metadata reference: C:\Users\chuon\RiderProjects\GlobalMEP\OpenMEPSandbox\OpenMEPSandbox.csproj
Processing OpenMEP
Processing OpenMEPSandbox
Creating output...
Completed Scope:MetadataCommand in 6171.3058 milliseconds.
Searching custom plugins in directory C:\Users\chuon\.dotnet\tools\.store\docfx\2.62.1\docfx\2.62.1\tools\net6.0\any\...
6 plug-in(s) loaded.
No files are found with glob pattern apidoc/**.md, excluding obj/**,_site/**, under directory "C:\Users\chuon\RiderProjects\GlobalMEP\docs"
Max parallelism is 8.
Unable to handle following files: api/toc.yml.bak. Do they miss `YamlMime` as the first line of file, e.g.: `### YamlMime:ManagedReference`?
Building 2 file(s) in TocDocumentProcessor(BuildTocDocument)...
Building 1 file(s) in ResourceDocumentProcessor(ValidateResourceMetadata)...
Building 70 file(s) in ManagedReferenceDocumentProcessor(BuildManagedReferenceDocument=>ValidateManagedReferenceDocumentMetadata=>ApplyOverwriteDocumentForMref=>FillReferenceInformatio
n)...
Building 2 file(s) in ConceptualDocumentProcessor(BuildConceptualDocument=>CountWord=>ValidateConceptualDocumentMetadata)...
1 external references found in 1 xref maps.
Applying templates to 75 model(s)...
XRef map exported.
Manifest file saved to manifest.json.
Completed building documents in 3249.5072 milliseconds.
Serving "C:\Users\chuon\RiderProjects\GlobalMEP\docs\_site" on http://localhost:8080. Press Ctrl+C to shut down.

```

Docs : 
![image](https://user-images.githubusercontent.com/31106432/223415304-f2ea2180-be8b-47d8-b78e-24c78ffcd974.png)
Folder after move : 
![image](https://user-images.githubusercontent.com/31106432/223415373-c2561c4d-7dde-4781-badd-b029d692c4fa.png)
Command :  docfx docs/docfx.json --serve

**Reply by chuongmep** [reply]
One more information : With insert .cs file, it's working normal, but image still same : 

![image](https://user-images.githubusercontent.com/31106432/223419496-d4acb02a-4c8c-406f-9eef-f94b091a4fb4.png)

```xml
/// <summary>
    /// return current version of dynamo
    /// </summary>
    /// <returns name="version">version</returns>
    /// <example>
    /// ![alt text](../../docs/OpenMEPPage/application/dyn/pic/Dynamo.Version.png "Dynamo")
    /// [!code-csharp[](../../docs/sample/navis.cs)]
    /// </example>
```

**Reply by chuongmep** [reply]
Hi @yufeih , after try snoop something in html file, I think I resolved my problem by short path, add string title can't generate image: 
 - Image : `![](../OpenMEPPage/application/dyn/pic/Dynamo.Version.png)`
![image](https://user-images.githubusercontent.com/31106432/223421989-1b94504c-eac8-4e1d-bac0-57fc0f932fbb.png)
